scripts/cicd/verify-cicd-operation.py

```diff
@@ -34,17 +38,54 @@ def runner(program, *args):
 logger.setLevel(logging.INFO)
 logger.addHandler(logging.handlers.SysLogHandler('/dev/log'))
 
-keyname = sys.argv[1] if len(sys.argv) >= 2 else "<none specified>"
+# The cluster_id argument is provided by the authorized_keys command entry for the ssh user's private key.
+# In other words, the ssh user cannot specify the cluster_id directly (they must have the correct private key
+# AND the key must be setup in the authorized_keys file appropriately).
+if len(sys.argv) != 2:
+    print("No cluster identifier specified")
+    sys.exit(12)
+
+cluster_id = sys.argv[1]
+
+# Sanity check the cluster_id from authorized_keys
+if not re.match("(^[a-zA-Z0-9.\-_]+$)", cluster_id):
```

Comment:
Minor : put the "-" at the end of the char ranges to not need to escape it.
Less minor : cluster names starting with a hyphen will complicate things terribly.

Suggest:   if not re.match("(^[a-zA-Z0-9][a-zA-Z0-9._-]+$)", cluster_id):